addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.onewiregpio.handler;
+
+import static org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OneWireGPIOHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class OneWireGPIOHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OneWireGPIOHandler.class);
+
+    private String gpioBusFile;
+    private Integer refreshTime;
+
+    public OneWireGPIOHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(TEMPERATURE)) {
+            if (command instanceof RefreshType) {
+                publishSensorValue(channelUID);
+            } else {
+                logger.debug("Command {} is not supported for channel: {}. Supported command: REFRESH", command,
+                        channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+        updateStatus(ThingStatus.INITIALIZING);
```

Comment:
Err, no, that's not what I meant.
If `checkConfiguration()` finds a problem, it set's the status to OFFLINE. You should therefore not set it to ANYTHING else here again.
If there is no config problem, it is perfectly fine to set it to UNKNOWN here as you did before.

Reply:
ok, so I removed the line

Reply:
Sorry, this is still not a correct solution.
You have to set the status to something within a few seconds. Not setting it here means that it needs to be set in your scheduled job. But looking at its implementation, I see that you will never set the status to anything, if no channel is linked.
The simplest solution is probably to return a boolean from checkConfiguration and simply not continue the processing, if there is a configuration error. Anyhow, you probably would not want to schedule the runnable either, right?